### PR TITLE
[BUG] Fix tsf data error log and make it more precise

### PR DESCRIPTION
**Pull request by pranavvp16**
Fixes error #6257

### CODE REVIEW

**Review by fkiraly — CHANGES_REQUESTED** [review]
Thanks!

While we are at it, could we:

* add a test, with `pytest.raises`? It is best practice to add a test for every bugfix (or, more generally, make sure the bug is covered).
* The error message optimally should mention the function or method in which it is raised, by name, and the argument. E.g., "error in xyz, invalid value for argument abc, etc"

You can also get the `bug` and `code` badges if you don't already have them.

File: sktime/datasets/_single_problem_loaders.py [hunk 1]
```diff
@@ -1413,10 +1413,8 @@ def load_forecastingdata(
         # valid dataset names for classification, regression, forecasting datasets repo
         if name not in list(tsf_all_datasets):
             raise ValueError(
-                {name}
-                + " is not a valid dataset name. \
-                    List of valid dataset names can be found at \
-                    sktime.datasets.tsf_dataset_names.tsf_all_datasets"
+                f"{name} is not a valid dataset name. "
+                f"List of valid datasets is {tsf_all_datasets}"
```
Comment (VascoSch92) [inline sktime/datasets/_single_problem_loaders.py:1417]:
```suggestion
                f"List of valid datasets is {list(tsf_all_datasets)}"
```
or?

**Comment by pranavvp16** [comment]
`ValueError: tourism-yearly-dataset is not a valid dataset name. List of valid dataset is ['m1_yearly_dataset', 'm1_quarterly_dataset', 'm1_monthly_dataset', 'm3_yearly_dataset', 'm3_quarterly_dataset', 'm3_monthly_dataset', 'm3_other_dataset', 'm4_yearly_dataset', 'm4_quarterly_dataset', 'm4_monthly_dataset', 'm4_weekly_dataset', 'm4_daily_dataset', 'm4_hourly_dataset', 'tourism_yearly_dataset', 'tourism_quarterly_dataset', 'tourism_monthly_dataset', 'cif_2016_dataset', 'london_smart_meters_dataset_with_missing_values', 'london_smart_meters_dataset_without_missing_values', 'australian_electricity_demand_dataset', 'wind_farms_minutely_dataset_with_missing_values', 'wind_farms_minutely_dataset_without_missing_values', 'dominick_dataset', 'bitcoin_dataset_with_missing_values', 'bitcoin_dataset_without_missing_values', 'pedestrian_counts_dataset', 'vehicle_trips_dataset_with_missing_values', 'vehicle_trips_dataset_without_missing_values', 'kdd_cup_2018_dataset_with_missing_values', 'kdd_cup_2018_dataset_without_missing_values', 'weather_dataset', 'nn5_daily_dataset_with_missing_values', 'nn5_daily_dataset_without_missing_values', 'nn5_weekly_dataset', 'kaggle_web_traffic_dataset_with_missing_values', 'kaggle_web_traffic_dataset_without_missing_values', 'kaggle_web_traffic_weekly_dataset', 'solar_10_minutes_dataset', 'solar_weekly_dataset', 'solar_4_seconds_dataset', 'electricity_hourly_dataset', 'electricity_weekly_dataset', 'car_parts_dataset_with_missing_values', 'car_parts_dataset_without_missing_values', 'fred_md_dataset', 'traffic_hourly_dataset', 'traffic_weekly_dataset', 'rideshare_dataset_with_missing_values', 'rideshare_dataset_without_missing_values', 'hospital_dataset', 'covid_deaths_dataset', 'sunspot_dataset_with_missing_values', 'sunspot_dataset_without_missing_values', 'saugeenday_dataset', 'us_births_dataset', 'wind_4_seconds_dataset']`

@fkiraly what error log should I keep. The current one looks like this

**Review by fkiraly — APPROVED** [review]
Thanks.

**Pull request merged**